src/hotspot/share/opto/phaseX.cpp

```diff
@@ -1160,6 +1184,813 @@ bool PhaseIterGVN::verify_node_value(Node* n) {
+    // trip the asserts later on.
+    //
+    // Found with:
+    //   compiler/eliminateAutobox/TestShortBoxing.java
+    //   -ea -esa -XX:CompileThreshold=100 -XX:+UnlockExperimentalVMOptions -server -XX:-TieredCompilation -XX:+IgnoreUnrecognizedVMOptions -XX:VerifyIterativeGVN=1110
+    return false;
+  }
+
+  if (n->is_CallJava()) {
+    // CallStaticJavaNode::Ideal
+    // Led to a crash:
+    //   assert((is_CallStaticJava() && cg->is_mh_late_inline()) || (is_CallDynamicJava() && cg->is_virtual_late_inline())) failed: mismatch
+    //
+    // Did not investigate yet, could be a bug.
+    // Or maybe it does not expect to be called during verification.
+    //
+    // Found with:
+    //   test/jdk/jdk/incubator/vector/VectorRuns.java
+    //   -XX:VerifyIterativeGVN=1110
+
+    // CallDynamicJavaNode::Ideal, and I think also for CallStaticJavaNode::Ideal
+    //  and possibly their subclasses.
+    // During late inlining it can call CallJavaNode::register_for_late_inline
+    // That means we do more rounds of late inlining, but might fail.
+    // Then we do IGVN again, and register the node again for late inlining.
+    // This creates an endless cycle. Everytime we try late inlining, we
+    // are also creating more nodes, especially SafePoint and MergeMem.
+    // These nodes are immediately rejected when the inlining fails in the
+    // do_late_inline_check, but they still grow the memory, until we hit
+    // the MemLimit and crash.
+    // The assumption here seems that CallDynamicJavaNode::Ideal does not get
+    // called repeatedly, and eventually we terminate. I fear this is not
+    // a great assumption to make. We should investigate more.
+    //
+    // Found with:
+    //   compiler/loopopts/superword/TestDependencyOffsets.java#vanilla-U
+    //   -XX:+IgnoreUnrecognizedVMOptions -XX:VerifyIterativeGVN=1110
+    return false;
+  }
+
+  // The number of nodes shoud not increase.
+  uint old_unique = C->unique();
+
+  Node* i = n->Ideal(this, can_reshape);
+  // If there was no new Idealization, we are probably happy.
+  if (i == nullptr) {
+    if (old_unique < C->unique()) {
+      tty->cr();
+      tty->print_cr("Ideal optimization did not make progress but created new unused nodes.");
+      tty->print_cr("  old_unique = %d, unique = %d", old_unique, C->unique());
+      n->dump_bfs(1, nullptr, "");
+      return true;
+    }
+
+    verify_empty_worklist(n);
+
+    // Everything is good.
+    return false;
+  }
+
+  // We just saw a new Idealization which was not done during IGVN.
+  tty->cr();
+  tty->print_cr("Missed Ideal optimization (can_reshape=%s):", can_reshape ? "true": "false");
+  if (i == n) {
+    tty->print_cr("The node was reshaped by Ideal.");
+  } else {
+    tty->print_cr("The node was replaced by Ideal.");
+    tty->print_cr("Old node:");
+    n->dump_bfs(1, nullptr, "");
+  }
+  tty->print_cr("The result after Ideal:");
+  i->dump_bfs(1, nullptr, "");
```

Comment:
Perhaps taking the tty lock might be appropriate, due to the amount of printing? Or do we know that nothing else is printing?

Reply:
You could also define a `stringStream ss` and pass that one to `dump_bfs()`. We do a similar thing for `print_ideal_ir()` to keep everything in a block. As a bonus: We don't suffer from a tty lock being broken - even though that would not affect correctness.